modules/geometry-calculation.js

```diff
@@ -2,13 +2,21 @@ const Line = require('./ecs6-class/line')
 const Point = require('./ecs6-class/point')
 
 const calculateDistance = (point1, point2) => {
+    if(!(point1 instanceof Point))
+        throw new Error('the type of point1 is not Point')
+    if(!(point2 instanceof Point))
+        throw new Error('the type of point2 is not Point')
     let distanceX = (point2.x - point1.x) ** 2;
     let distanceY = (point2.y - point1.y) ** 2;
     const distance = Math.sqrt(distanceX + distanceY);
     return distance;
 }
 
 const calculateJunctionPoint = (line1, line2) => {
+    if(!(line1 instanceof Line))
+        throw new Error('the type of line1 is not Line')
+    if(!(line2 instanceof Line))
+        throw new Error('the type of line2 is not Line')
     if (line1.slope === line2.slope) {
```

Comment:
What happens if the slope or n are undefined, in one of the lines or both?

Reply:
I fixed it.